app/models/package_manager/maven/maven_central.rb

```diff
@@ -16,6 +16,32 @@ def self.missing_version_remover
     PackageManager::Base::MissingVersionRemover
   end
 
+  def self.latest_version(name)
+    metadata_version = super
+
+    # When maven-metadata.xml returned an interpolation string for latest version, e.g. ${revision},
+    # attempt to scrape MavenCentral's index HTML to infer the latest version.
+    if metadata_version =~ /\$\{/
+      version_listing = get_html(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).base)
+      scraped_version = version_listing
+        .css("#contents a")                                  # scrape the list of file/folders
+        .map(&:text)                                         # get each innerText
+        .select { |text| text.end_with?("/") }               # only look at folders
+        .map { |folder| folder.chomp("/") }                  # remove folder trailing slash
+        .grep(/^\d+.\d/)                                     # only folders that look like versions
+        .max_by do |text|
+          # Maven versions range from 1 to many "." and may not be valid SemVer. Use the more forgiving Gem::Version to sort
```

Comment:
I took a sample of 30k latest_versions for Maven packages, and found a pretty wide spectrum of version digits ("2" being SemVer-compliant for most cases):

```
> latest_versions.map { |v| v.count(".") }.tally
=> {1=>6704, 2=>18558, 0=>332, 3=>2591, 4=>71, 6=>1, 5=>13
```